File: template-method/java/HtmlDataAnalyzer.java
package TemplateMethod;

// UML diyagramındaki AbstractClass (DataAnalyzerBase) sınıfından türer.
// UML diyagramındaki ConcreteClass yapısına denk gelmektedir.
class HtmlDataAnalyzer extends DataAnalyzerBase {
    @Override
    public void exportResult(Object result) {
        // Durumun HTML'e dönüştürülmesi işlemleri yer alacaktır.
        System.out.println("HTML exported.");
    }
}

File: template-method/kotlin/HtmlDataAnalyzer.kt
package TemplateMethod

// UML diyagramındaki AbstractClass (DataAnalyzerBase) sınıfından türer.
// UML diyagramındaki ConcreteClass yapısına denk gelmektedir.
internal class HtmlDataAnalyzer : DataAnalyzerBase() {
    override fun exportResult(result: Any) {
        // Durumun HTML'e dönüştürülmesi işlemleri yer alacaktır.
        println("HTML exported.")
    }
}
